pkg/config/setup/config_test.go

```diff
@@ -1100,3 +1099,108 @@ func TestProxyLoadedFromConfigFileAndEnvVars(t *testing.T) {
 	assert.Equal(t, proxyHTTPEnvVar, proxyHTTPConfig)
 	assert.Equal(t, proxyHTTPSEnvVar, proxyHTTPSConfig)
 }
+
+var testExampleConf = []byte(`
+secret_backend_command: some command
+additional_endpoints:
+  https://url1.com:
+    - first
+    - second
+  https://url2.eu:
+    - third
+process_config:
+  additional_endpoints:
+    https://url1.com:
+      - fourth
+      - fifth
+    https://url2.eu:
+      - sixth
+`)
+
+func TestConfigAssignAtPath(t *testing.T) {
+	// CircleCI sets NO_PROXY, so unset it for this test
+	unsetEnvForTest(t, "NO_PROXY")
+
+	config := Conf()
+	config.SetWithoutSource("use_proxy_for_cloud_metadata", true)
+	configPath := filepath.Join(t.TempDir(), "datadog.yaml")
+	os.WriteFile(configPath, testExampleConf, 0600)
+	config.SetConfigFile(configPath)
+
+	_, err := LoadCustom(config, "unit_test", optional.NewNoneOption[secrets.Component](), nil)
+	assert.NoError(t, err)
+
+	err = configAssignAtPath(config, []string{"secret_backend_command"}, "different")
+	assert.NoError(t, err)
+
+	err = configAssignAtPath(config, []string{"additional_endpoints", "https://url1.com", "1"}, "changed")
+	assert.NoError(t, err)
+
+	err = configAssignAtPath(config, []string{"process_config", "additional_endpoints", "https://url2.eu", "0"}, "modified")
+	assert.NoError(t, err)
+
+	expectedYaml := `additional_endpoints:
+  https://url1.com:
+  - first
+  - changed
+  https://url2.eu:
+  - third
+process_config:
+  additional_endpoints:
+    https://url1.com:
+    - fourth
+    - fifth
+    https://url2.eu:
+    - modified
+secret_backend_command: different
+use_proxy_for_cloud_metadata: true
+`
+	yamlConf, err := yaml.Marshal(config.AllSettingsWithoutDefault())
+	assert.NoError(t, err)
+	yamlText := string(yamlConf)
+	assert.Equal(t, expectedYaml, yamlText)
+
+	err = configAssignAtPath(config, []string{"0"}, "invalid")
+	assert.Error(t, err)
+	assert.Equal(t, err.Error(), "invalid config setting '[0]'")
+
+	err = configAssignAtPath(config, []string{"additional_endpoints", "https://url1.com", "5"}, "invalid")
+	assert.Error(t, err)
+	assert.Equal(t, err.Error(), "index out of range 5 >= 2")
+}
+
+func TestConfigAssignAtPathForIntMapKeys(t *testing.T) {
+	// CircleCI sets NO_PROXY, so unset it for this test
+	unsetEnvForTest(t, "NO_PROXY")
```

Comment:
Could we explain why ENV var modifies the behaviour of the test?

Reply:
I have to admit, I mostly just copy and pasted this code from other tests, and so don't totally understand it. Without this line, the proxy section of the config is filled out, but the details within depend upon the current machine, so the test can't be written in a way that passes on CI. I think it would be good to fix this situation, but that may be out of scope for this PR.

Reply:
Thanks for the explanation. 

> I think it would be good to fix this situation, but that may be out of scope for this PR.

I totally agree with that 😄